astropy/io/misc/yaml.py

```diff
@@ -223,6 +243,23 @@ def ignore_aliases(self, data):
 AstropyDumper.add_representer(TimeDelta, _timedelta_representer)
 AstropyDumper.add_representer(coords.SkyCoord, _skycoord_representer)
 
+# Numpy dtypes
+AstropyDumper.add_representer(np.bool_,
+                              yaml.representer.SafeRepresenter.represent_bool)
+for np_type in [np.int_, np.int, np.intc, np.intp, np.int8, np.int16, np.int32,
+                np.int64, np.uint8, np.uint16, np.uint32, np.uint64]:
+    AstropyDumper.add_representer(np_type,
+                                 yaml.representer.SafeRepresenter.represent_int)
+for np_type in [np.float_, np.float, np.float16, np.float32, np.float64,
+                np.longdouble]:
```

Comment:
@matteobachetti - ok on longdouble and complex128, thanks for explanation.

I believe that pyyaml / yaml saves in a text form that fully represents at least float64 and int64 values since it outputs the `repr` (as opposed to `print` that uses `str`).  I don't know about the representation of `longdouble`, but since it will be constructed with a Python float I expect it will lose precision.  I guess this is not the end of the world...